### DISCUSSION TITLE
How to create an inline keyboard

### DISCUSSION THREAD

**Discussion by asteriskDevelop** [telegram-bot-sdk/telegram-bot-sdk#56] [Q&A]
How to create an inline keyboard, version dev-master(4.x-dev)
Help please

**Comment by brevisnguyen** [comment]
`$msg = Telegram::bot()->sendMessage([
    'chat_id' => $sender->id,
   'text' => 'Select option',
    'reply_markup' => json_encode([
        'inline_keyboard' => [ ['text' => 'Yes', 'callback_data' => 'yes'], ['text' => 'No', 'callback_data' => 'no'] ]
    ])
])`

Hope help you!

**Comment by alies-dev** [comment]
**OOP way**

```php
use Telegram\Bot\Objects\Keyboard\InlineKeyboardButton;
use Telegram\Bot\Objects\Keyboard\InlineKeyboardMarkup;

// where:
// $bot - \Telegram\Bot\Bot::class instance
// $update - \Telegram\Bot\Objects\Update::class instance

$bot->sendMessage([
    'chat_id' => $update->getChat()->id,
    'text' => 'How are you?',
    'reply_markup' => (new InlineKeyboardMarkup())
        ->row(new InlineKeyboardButton(['text' => 'Not bad', 'callback_data' => 'bot_bad']))
        ->row(new InlineKeyboardButton(['text' => 'Good', 'callback_data' => 'good']));
```

**Reply by brevisnguyen** [reply]
> **OOP way**
> 
> ```
> use Telegram\Bot\Objects\Keyboard\InlineKeyboardButton;
> use Telegram\Bot\Objects\Keyboard\InlineKeyboardMarkup;
> 
> // where:
> // $bot - \Telegram\Bot\Bot::class instance
> // $update - \Telegram\Bot\Objects\Update::class instance
> 
> $bot->sendMessage([
>     'chat_id' => $update->getChat()->id,
>     'text' => 'How are you?',
>     'reply_markup' => (new InlineKeyboardMarkup())
>         ->row(new InlineKeyboardButton(['text' => 'Not bad', 'callback_data' => 'bot_bad']))
>         ->row(new InlineKeyboardButton(['text' => 'Good', 'callback_data' => 'good']));
> ```

How to handle callback_data?

**Reply by alies-dev** [reply]
@brevis-ng 
```php
if ($update->callback_query?->data === 'YOUR_CB_DATA') {
    $bot->sendMessage([
        'chat_id' => $update->getChat()->id,
        'text' => 'Hello',
    ]);
}

```

**Reply by mauron-ciandt** [reply]
where would I check for that @lptn ? in the default endpoint that handles the webhook call?

**Reply by alies-dev** [reply]
@mauron-ciandt 
Wherever the `Telegram\Bot\Objects\Update` instance available. My setup is different - I used [`koot-labs/telegram-bot-dialogs` ](https://github.com/koot-labs/telegram-bot-dialogs) package where instances of this object available in all dialog steps
